superset-frontend/src/dashboard/components/FiltersBadge/DetailsPanel.tsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { useTheme } from '@superset-ui/core';
+import {
+  SearchOutlined,
+  MinusCircleFilled,
+  CheckCircleFilled,
+  ExclamationCircleFilled,
+} from '@ant-design/icons';
+import { Collapse } from 'src/common/components/index';
+import S from './Styles';
+import { APPLIED, INCOMPATIBLE, UNSET } from './selectors';
+
+export type Indicator = {
+  id: string;
+  name: string;
+  value: string[];
+  status: typeof APPLIED | typeof UNSET | typeof INCOMPATIBLE;
```

Comment:
You can just create a custom type `IndicatorStatus = 'applied' | 'unset' | 'incompatible'` and use string literals when comparing against `Indicator.status`. TypeScript will be smart enough to give you errors and autocomplete hints.

I would also recommend naming `Indicator` as `FilterIndicator` to make things more explicit and easier to understand---just imagine we are going to add another indicator for something else in the future. Types should always be assumed to be globals since they tend to be imported here and there once being exposed via some API.

```typescript
export type FilterIndicatorStatus = 'applied' | 'unset' | 'incompatible'

export type FilterIndicator = {
  ...
  status: FilterIndicatorStatus;
}
```

Reply:
I changed the statuses from consts to an enum. Cleaner all 'round.

Disagree on renaming. This file is in the `FiltersBadge` directory so its relation to filtering should be apparent.